### PR TITLE
refactor(breadcrumb): use new step color tokens

### PR DESCRIPTION
**Pull request by mapsandapps**
Issue number: Internal

---------

## What is the current behavior?

The breadcrumbs use the `--ion-color-step` tokens or a fallback.

## What is the new behavior?

- The breadcrumb text colors use the `--ion-color-step` tokens or the `--ion-text-color-step` tokens or a fallback
- The breadcrumb background colors use the `--ion-color-step` tokens or the `--ion-background-color-step` tokens or a fallback

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/breadcrumb/breadcrumb.ios.vars.scss [hunk 1]
```diff
@@ -4,22 +4,22 @@
...
 /// @prop - Color of the breadcrumb
-$breadcrumb-ios-color:                             var(--ion-color-step-850, #2d4665) !default;
+$breadcrumb-ios-color:                             var(--ion-color-step-850, var(--ion-text-color-step-850, #2d4665)) !default;
 
 /// @prop - Color of the active breadcrumb
 $breadcrumb-ios-color-active:                      var(--ion-text-color, #03060b) !default;
 
 /// @prop - Background color of the focused breadcrumb
-$breadcrumb-ios-background-focused:                var(--ion-color-step-50, rgba(233, 237, 243, 0.7)) !default;
+$breadcrumb-ios-background-focused:                var(--ion-color-step-50, var(--ion-background-color-step-50,rgba(233, 237, 243, 0.7))) !default;
```
Comment (amandaejohnston) [inline core/src/components/breadcrumb/breadcrumb.ios.vars.scss:13]:
```suggestion
$breadcrumb-ios-background-focused:                var(--ion-color-step-50, var(--ion-background-color-step-50, rgba(233, 237, 243, 0.7))) !default;
```

File: core/src/components/breadcrumb/breadcrumb.ios.vars.scss [hunk 2]
```diff
@@ -4,22 +4,22 @@
 // --------------------------------------------------
 
 /// @prop - Color of the breadcrumb
-$breadcrumb-ios-color:                             var(--ion-color-step-850, #2d4665) !default;
+$breadcrumb-ios-color:                             var(--ion-color-step-850, var(--ion-text-color-step-850, #2d4665)) !default;
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/breadcrumb/breadcrumb.ios.vars.scss:7]: Your understanding is correct. The legacy stepped colors go from white (the background) to black (the text color). However, when using the _text_ step colors those always go from black (the text color) to white (the background) when in light mode. That's why we need to invert the numbers for the text color tokens only.

**Comment by mapsandapps** [comment]
Ok, I think this is finally good to go. We decided that

* None of the `--ion-color-step` values should change
* Text step colors should be equal to 1000 minus the `--ion-color-step` value
* Even though most non-text things should use the background colors, icons that are inline with text like this should use the text colors

**Review by liamdebeasi — APPROVED** [review]
Thank you!

**Review by brandyscarney — APPROVED** [review]

**Pull request merged**